include/wait_strategy.hpp

```diff
@@ -0,0 +1,379 @@
+            counter = waitMethod();
+        }
+
+        return availableSequence;
+    }
+};
+static_assert(WaitStrategy<SleepingWaitStrategy>);
+static_assert(!hasSignalAllWhenBlocking<SleepingWaitStrategy>);
+
+struct TimeoutException : public std::runtime_error {
+    TimeoutException() : std::runtime_error("TimeoutException") {}
+};
+
+class TimeoutBlockingWaitStrategy {
+    using Clock = std::conditional_t<std::chrono::high_resolution_clock::is_steady, std::chrono::high_resolution_clock, std::chrono::steady_clock>;
+    Clock::duration             _timeout;
+    std::recursive_mutex        _gate;
+    std::condition_variable_any _conditionVariable;
+
+public:
+    explicit TimeoutBlockingWaitStrategy(Clock::duration timeout)
+        : _timeout(timeout) {}
+
+    std::int64_t waitFor(const std::int64_t sequence, const Sequence &cursor, const std::vector<std::shared_ptr<Sequence>> &dependentSequences) {
+        auto timeSpan = std::chrono::microseconds(std::chrono::duration_cast<std::chrono::microseconds>(_timeout).count());
+
+        if (cursor.value() < sequence) {
+            std::unique_lock uniqueLock(_gate);
+
+            while (cursor.value() < sequence) {
+                // optional: barrier check alert
+
+                if (_conditionVariable.wait_for(uniqueLock, timeSpan) == std::cv_status::timeout) {
+                    throw TimeoutException();
+                }
+            }
+        }
+
+        std::int64_t availableSequence;
+        while ((availableSequence = detail::getMinimumSequence(dependentSequences)) < sequence) {
+            // optional: barrier check alert
+        }
+
+        return availableSequence;
+    }
+
+    void signalAllWhenBlocking() {
+        std::unique_lock uniqueLock(_gate);
+        _conditionVariable.notify_all();
+    }
+};
+static_assert(WaitStrategy<TimeoutBlockingWaitStrategy>);
+static_assert(hasSignalAllWhenBlocking<TimeoutBlockingWaitStrategy>);
+
+/**
+ * Yielding strategy that uses a Thread.Yield() for IEventProcessors waiting on a barrier after an initially spinning.
+ * This strategy is a good compromise between performance and CPU resource without incurring significant latency spikes.
+ */
+class YieldingWaitStrategy {
+    const std::size_t _spinTries = 100;
+
+public:
+    std::int64_t waitFor(const std::int64_t sequence, const Sequence & /*cursor*/, const std::vector<std::shared_ptr<Sequence>> &dependentSequences) const {
+        auto       counter    = _spinTries;
+        const auto waitMethod = [&]() {
+            // optional: barrier check alert
+
+            if (counter == 0) {
+                std::this_thread::yield();
+            } else {
+                --counter;
+            }
+            return counter;
+        };
+
+        std::int64_t availableSequence;
+        while ((availableSequence = detail::getMinimumSequence(dependentSequences)) < sequence) {
+            counter = waitMethod();
```

Comment:
Lambda takes counter by reference, so the `=` is a noop

Reply:
removed the superfluous return and `counter =` assignment.